### ISSUE TITLE
client & cache diagram

### ISSUE THREAD

**Issue by andreineculau** [for-GET/http-decision-diagram#14]
it is not only the server that can benefit from an FSM to decide the HTTP response flow,
but also the client ~~and the cache proxy (a version is already in master branch based on @darrelmiller's versions)~~

trigger: http://blog.safaribooksonline.com/2013/05/31/rest-from-a-client-perspective/

**Comment by pke** [comment]
I agree that a client http decision diagram would be helpful and maybe a reference implementation in JavaScript using xstate?

**Comment by andreineculau** [comment]
@pke thanks for your interest and your comment! Had no idea about https://github.com/davidkpiano/xstate .
I have used https://github.com/ifandelse/machina.js/ 7 years ago.
And it looks rather easy to parse cosmogol into xstate.
